cfme/utils/blockers.py

```diff
@@ -181,6 +181,44 @@ def bugzilla_bug(self):
             return None
         return self.data
 
+    @property
+    def can_test_on_current_upstream_appliance(self):
+        """
+        This property is designed for use with the blocks property and is answering the question:
+
+        "If I have a bug, and that bug is in a "POST" or "MODIFIED" state, and if I have an
+        upstream appliance, is the build date of my current upstream appliance after
+        the fix for the bug was merged?"
+
+        If so, then it will return True, because the bug's fix is included in your current
+        upstream appliance.
+
+        If you happen to be on an older version of an upstream appliance that doesn't have the fix,
+        then it will return False.
+        """
+        bug = self.data
+        if bug is None:  # if bug is None, then it is not a blocker and we can test it
+            return True
+        if not bug.can_test_on_upstream:  # if bug is not fixed in upstream, we can't test it!
+            return False
+
+        change_states = {"POST", "MODIFIED"}
+        # With these states, the change is not in upstream
+        if bug.status in {"NEW", "ON_DEV", "ASSIGNED"}:
+            return False
+        history = bug.get_history_raw()["bugs"][0]["history"]
+        changes = []
+        # We look for status changes in the history
+        for event in history:
+            for change in event["changes"]:
+                if change["field_name"].lower() != "status":
+                    continue
+                if change["added"] in change_states:
+                    changes.append(event["when"])
+                    return event["when"] < version.appliance_build_datetime()
```

Comment:
I'm hitting a `TypeError` here because `version.appliance_build_datetime()` is returning `None`.

Creating an appliance object using `Appliance.from_provider`, with a MIQ nightly build, results in an appliance object with Version('master'), which has no build datetime.

Reply:
Moving to WIPTEST until this is resolved

Reply:
This should be fixed by a small change to the `build` property of an appliance object